### PR TITLE
Em 82

### PR DESCRIPTION
**Pull request by matrix-drorgo**
Not yet finished ,Waiting for server side to result this solution.

image

1) missing names of performers .
2) sorting of the hours

### CODE REVIEW

File: src/Components/Forms/TestsAndTreatments/TestsAndTreatments.js [hunk 1]
```diff
@@ -1,18 +1,201 @@
...
+        'Pain level': null,
+        'Blood sugar': null,
+      },
+    ],
+  );
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const clinicIndicators = await getIndicatorsSettings();
```
[the remark this answers is not in the document]
Comment (matrix-amiel) [reply src/Components/Forms/TestsAndTreatments/TestsAndTreatments.js:58]: you can help with Idan's code Components/Generic/PatientAdmission/PatientDetailsBlock/index.js:258

[the remark this answers is not in the document]
Comment (matrix-drorgo) [reply src/Components/Forms/TestsAndTreatments/TestsAndTreatments.js:58]: What ?

[the remark this answers is not in the document]
Comment (matrix-drorgo) [reply src/Components/Forms/TestsAndTreatments/TestsAndTreatments.js:58]: sorry But I don't understand you !!!

File: src/Components/Generic/MaskedControllers/FormattedInputs/FormattedInputs.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (matrix-drorgo) [reply src/Components/Generic/MaskedControllers/FormattedInputs/FormattedInputs.js:1]: This is a generic code that every one can use why did you decide that only test and treatments can use it ?

File: src/Utils/Services/FHIR/Observations/index.js [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    return CRUDOperations(
+      'search',
+      `${params.url}?${patient ? `patient=${patient}` : ''}${
+        encounter ? `&encounter=${encounter}` : ''
+      }${category ? `&category=${category}` : ''}${
+        include ? `&_include=${include}` : ''
+      }${sort ? `&_sort=${sort}` : ''}`,
+    );
+
+    return {
```
[the remark this answers is not in the document]
Comment (matrix-drorgo) [reply src/Utils/Services/FHIR/Observations/index.js:37]: As we talked you wanted to look at this branch , this PR is not yet done .... I think you should wait as I need to clean the code and arrange the classes path and all the things ....

File: src/Components/Generic/MaskedControllers/FormattedInputs/FormattedInputs.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (matrix-amiel) [reply src/Components/Generic/MaskedControllers/FormattedInputs/FormattedInputs.js:1]: ok... I see.. so it's kind of helper, let's move it to Helpers folder

File: src/Components/Forms/TestsAndTreatments/TestsAndTreatments.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (matrix-drorgo) [reply src/Components/Forms/TestsAndTreatments/TestsAndTreatments.js:58]: I have changed it but I don't see the time issue you talked about it is the same time but now all object are drawn to the screen in one iteration , not as before As **I wanted** it to be they were drawn one by one . so visually I think it is a mistake to drop to the client many object once and not one by one but you know  I have already changed it.

File: src/Components/Generic/MaskedControllers/FormattedInputs/FormattedInputs.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (matrix-drorgo) [reply src/Components/Generic/MaskedControllers/FormattedInputs/FormattedInputs.js:1]: as we talked I have moved it to clinikal-react/src/Assets/Elements/MaskedControllers as it is not an helper it is a generic UI component maybe an asset ....

**Comment by matrix-drorgo** [comment]
Done...

File: src/Components/Forms/TestsAndTreatments/TestsAndTreatments.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (matrix-amiel) [reply src/Components/Forms/TestsAndTreatments/TestsAndTreatments.js]: I merge it now as is and we will talk and think about it.
I think the best practice is to hide the screen (even with loader) until all the elements are shown

[the remark this answers is not in the document]
Comment (matrix-drorgo) [reply src/Components/Forms/TestsAndTreatments/TestsAndTreatments.js]: before it was o.k. every part fills the screen in it's own pace .never-mind ...

**Pull request merged**